cmd/oras/push.go

```diff
@@ -125,16 +126,18 @@ func runPush(opts pushOptions) error {
 	// bake artifact
 	var pushOpts []oras.PushOpt
 	if opts.artifactType != "" {
-		resolver, err = orasDocker.WithDiscover(opts.artifactRefs, resolver, orasDocker.NewOpts(ropts))
-		if err != nil {
-			return err
-		}
-
-		manifest, err := loadReference(ctx, resolver, opts.artifactRefs)
-		if err != nil {
+		var manifest ocispec.Descriptor
+		var replacement remotes.Resolver
+		replacement, err = orasDocker.WithDiscover(opts.artifactRefs, resolver, orasDocker.NewOpts(ropts))
+		if err == nil {
+			resolver = replacement
+			manifest, err = loadReference(ctx, resolver2, opts.artifactRefs)
```

Comment:
This is wrong, not sure why this is what was pushed.  Will clean-up and also consider not calling discover at all, rather than call and rely on it failing with the hostname error.